### ISSUE TITLE
docker-machine & 1.1.0: shutdown regression (timeout then hard shutdown)

### ISSUE THREAD

**Issue by ryansch** [rancher/os#2091]
**RancherOS Version: (ros os version)**
v1.0.4, v1.1.0

**Where are you running RancherOS? (docker-machine, AWS, GCE, baremetal, etc.)**
docker-machine

I can't seem to stop v1.1.0 via docker-machine or Parallels in a timely fashion.  It appears that it only works because it hits some kind of internal timeout and hard powers off the VM.  This has caused some interesting corruption issues.

Here are back to back create and stops with v1.0.4 and v1.1.0:

```sh
$ docker-machine create -d parallels --parallels-memory 2048 --parallels-cpu-count 4 --parallels-disk-size 60000 --parallels-no-share --parallels-boot2docker-url https://releases.rancher.com/os/v1.0.4/rancheros.iso test104
Running pre-create checks...
Creating machine...
(test104) Downloading /Users/ryansch/.docker/machine/cache/boot2docker.iso from https://releases.rancher.com/os/v1.0.4/rancheros.iso...
(test104) 0%....10%....20%....30%....40%....50%....60%....70%....80%....90%....100%
(test104) Creating SSH key...
(test104) Creating Parallels Desktop VM...
(test104) Starting Parallels Desktop VM...
(test104) Waiting for VM to come online...
Waiting for machine to be running, this may take a few minutes...
Detecting operating system of created instance...
Waiting for SSH to be available...
Detecting the provisioner...
Provisioning with rancheros...
Copying certs to the local machine directory...
Copying certs to the remote machine...
Setting Docker configuration on the remote daemon...
Checking connection to Docker...
Docker is up and running!
To see how to connect your Docker Client to the Docker Engine running on this virtual machine, run: docker-machine env test104

$ time docker-machine stop test104
Stopping "test104"...
Machine "test104" was stopped.
docker-machine stop test104  0.01s user 0.01s system 0% cpu 5.521 total

$ docker-machine create -d parallels --parallels-memory 2048 --parallels-cpu-count 4 --parallels-disk-size 60000 --parallels-no-share --parallels-boot2docker-url https://releases.rancher.com/os/v1.1.0/rancheros.iso test110
Running pre-create checks...
Creating machine...
(test110) Downloading /Users/ryansch/.docker/machine/cache/boot2docker.iso from https://releases.rancher.com/os/v1.1.0/rancheros.iso...
(test110) 0%....10%....20%....30%....40%....50%....60%....70%....80%....90%....100%
(test110) Creating SSH key...
(test110) Creating Parallels Desktop VM...
(test110) Starting Parallels Desktop VM...
(test110) Waiting for VM to come online...
Waiting for machine to be running, this may take a few minutes...
Detecting operating system of created instance...
Waiting for SSH to be available...
Detecting the provisioner...
Provisioning with rancheros...
Copying certs to the local machine directory...
Copying certs to the remote machine...
Setting Docker configuration on the remote daemon...
Checking connection to Docker...
Docker is up and running!
To see how to connect your Docker Client to the Docker Engine running on this virtual machine, run: docker-machine env test110

$ time docker-machine stop test110
Stopping "test110"...
Machine "test110" was stopped.
docker-machine stop test110  0.01s user 0.01s system 0% cpu 2:01.25 total
```

**Comment by SvenDowideit** [comment]
I don't have parallels - so this will be "interesting" (atm, i don't even have a mac)

**Comment by ryansch** [comment]
@SvenDowideit I don't _think_ this is a parallels specific bug.  I'll try to repro on another virtualization platform.

**Comment by SvenDowideit** [comment]
yeah, quite plausible - I don't use `docker-machine stop` much - I'll have to add it to my list of things to test

**Comment by dx9** [comment]
hi guys, I'm getting this with the xhyve driver too

**Comment by ryansch** [comment]
@SvenDowideit Any way we could get this into 1.1.1?  We're blocked on upgrading until it's fixed.  Any other information I can supply?

**Comment by daiyam** [comment]
same here with virtualbox

```
$ docker-machine create -d virtualbox --virtualbox-boot2docker-url https://releases.rancher.com/os/v1.0.4/rancheros.iso test104
Running pre-create checks...
(test104) Boot2Docker URL was explicitly set to "https://releases.rancher.com/os/v1.0.4/rancheros.iso" at create time, so Docker Machine cannot upgrade this machine to the latest version.
Creating machine...
(test104) Boot2Docker URL was explicitly set to "https://releases.rancher.com/os/v1.0.4/rancheros.iso" at create time, so Docker Machine cannot upgrade this machine to the latest version.
(test104) Downloading /Users/baptiste/.docker/machine/cache/boot2docker.iso from https://releases.rancher.com/os/v1.0.4/rancheros.iso...
(test104) 0%....10%....20%....30%....40%....50%....60%....70%....80%....90%....100%
(test104) Creating VirtualBox VM...
(test104) Creating SSH key...
(test104) Starting the VM...
(test104) Check network to re-create if needed...
(test104) Waiting for an IP...
Waiting for machine to be running, this may take a few minutes...
Detecting operating system of created instance...
Waiting for SSH to be available...
Detecting the provisioner...
Provisioning with rancheros...
Copying certs to the local machine directory...
Copying certs to the remote machine...
Setting Docker configuration on the remote daemon...
Checking connection to Docker...
Docker is up and running!
To see how to connect your Docker Client to the Docker Engine running on this virtual machine, run: docker-machine env test104

$ time docker-machine stop test104
Stopping "test104"...
Machine "test104" was stopped.

real	0m10.906s
user	0m0.019s
sys	0m0.029s

$ docker-machine create -d virtualbox --virtualbox-boot2docker-url https://releases.rancher.com/os/v1.1.0/rancheros.iso test110
Running pre-create checks...
(test110) Boot2Docker URL was explicitly set to "https://releases.rancher.com/os/v1.1.0/rancheros.iso" at create time, so Docker Machine cannot upgrade this machine to the latest version.
Creating machine...
(test110) Boot2Docker URL was explicitly set to "https://releases.rancher.com/os/v1.1.0/rancheros.iso" at create time, so Docker Machine cannot upgrade this machine to the latest version.
(test110) Downloading /Users/baptiste/.docker/machine/cache/boot2docker.iso from https://releases.rancher.com/os/v1.1.0/rancheros.iso...
(test110) 0%....10%....20%....30%....40%....50%....60%....70%....80%....90%....100%
(test110) Creating VirtualBox VM...
(test110) Creating SSH key...
(test110) Starting the VM...
(test110) Check network to re-create if needed...
(test110) Waiting for an IP...
Waiting for machine to be running, this may take a few minutes...
Detecting operating system of created instance...
Waiting for SSH to be available...
Detecting the provisioner...
Provisioning with rancheros...
Copying certs to the local machine directory...
Copying certs to the remote machine...
Setting Docker configuration on the remote daemon...
Checking connection to Docker...
Docker is up and running!
To see how to connect your Docker Client to the Docker Engine running on this virtual machine, run: docker-machine env test110

$ time docker-machine stop test110
Stopping "test110"...
^C

real	240m44.626s
user	0m14.816s
sys	0m7.480s
```

**Comment by frickenate** [comment]
Same issue. I'm using VirtualBox. Upon attempting the `docker-machine stop` command, here are the 6 lines immediately written to `/var/log/system-docker.log` within the VM:

`time="2017-11-14T22:44:19.304128690Z" level=warning msg="Could not get operating system name: Error opening /usr/lib/os-release: open /usr/lib/os-release: no such file or directory"`

`time="2017-11-14T22:44:19.403997579Z" level=error msg="Handler for GET /containers/sbin/poweroff/json returned error: No such container: sbin/poweroff"`

`time="2017-11-14T22:44:19.406781661Z" level=error msg="Handler for POST /containers/create returned error: Invalid container name (/sbin/poweroff), only [a-zA-Z0-9][a-zA-Z0-9_.-] are allowed"`

`time="2017-11-14T22:44:19.459887346Z" level=warning msg="Could not get operating system name: Error opening /usr/lib/os-release: open /usr/lib/os-release: no such file or directory"`

`time="2017-11-14T22:44:19.560547123Z" level=error msg="Handler for GET /containers/sbin/poweroff/json returned error: No such container: sbin/poweroff"`

`time="2017-11-14T22:44:19.563329985Z" level=error msg="Handler for POST /containers/create returned error: Invalid container name (/sbin/poweroff), only [a-zA-Z0-9][a-zA-Z0-9_.-] are allowed"`

If there's a timeout after which a forced poweroff is supposed to happen, I haven't seen that after waiting for 5 minutes.

**Comment by ryansch** [comment]
I've confirmed this is fixed on 1.1.2.
